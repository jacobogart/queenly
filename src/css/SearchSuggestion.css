p {
  font-family: sans-serif;
  font-size: 20px;
  color: black;
  text-align: left; 
}