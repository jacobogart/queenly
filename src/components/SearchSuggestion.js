import React, { Component } from "react";
import "../css/SearchSuggestion.css";

class SearchSuggestion extends Component {
  constructor(props) {
    super(props);
  }

  setResult = () => {
    this.props.selectSearchResult(this.props.name);
  };

  render() {
    return (
      <div className="SuggestionsContainer">
        <p onClick={this.setResult}>{this.props.name}</p>
      </div>
    );
  }
}

export default SearchSuggestion;
