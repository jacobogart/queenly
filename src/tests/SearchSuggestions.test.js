import React from "react";
import SearchSuggestion from "../components/SearchSuggestion";
import { shallow } from "enzyme";

const mockSelectSearchResult = jest.fn();
const mockResult = {
  category: "viewing party",
  dayOfWeek: "Thursday",
  frequency: "weekly",
  host: ["Dixie Lynn Cartwright", "DiDa Ritz"],
  id: 1201,
  reoccuring: true,
  startTime: 2000,
  name: "Chicago's RuPaul's Drag Race Season 11 Viewing Party"
};

describe("SearchSuggestion", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SearchSuggestion
        name={mockResult.name}
        selectSearchResult={mockSelectSearchResult}
        key={mockResult.id}
      />
    );
  });

  it("should match the snapshot with all the data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should on click set the result to be the prop.name clicked", () => {
    let setResultSpy = jest.spyOn(wrapper.instance(), 'setResult');
    wrapper.instance().setResult()
    expect(setResultSpy).toHaveBeenCalled();
    expect(mockSelectSearchResult).toHaveBeenCalledWith(
      "Chicago's RuPaul's Drag Race Season 11 Viewing Party"
    );
  });
});
