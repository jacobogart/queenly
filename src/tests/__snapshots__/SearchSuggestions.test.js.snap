// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSuggestion should match the snapshot with all the data passed in 1`] = `undefined`;
